File: tmp_rovodev_generate_icons.py
#!/usr/bin/env python3
"""
CineVault Icon Generator
Converts the SVG icon to all required formats using cairosvg and Pillow
"""

try:
    import cairosvg
    from PIL import Image
    import io
    import os
except ImportError:
    print("❌ Required packages not found.")
    print("\nPlease install:")
    print("  pip install cairosvg pillow")
    print("\nOr use the Node.js version:")
    print("  npm run generate-icons")
    exit(1)

SVG_PATH = "src-tauri/icons/icon.svg"
ICONS_DIR = "src-tauri/icons"

# Icon sizes to generate
SIZES = [
    (32, "32x32.png"),
    (128, "128x128.png"),
    (256, "128x128@2x.png"),
    (512, "icon.png"),
    (1024, "icon@2x.png"),
]

def generate_png(svg_path, output_path, size):
    """Generate a PNG from SVG at specified size"""
    png_data = cairosvg.svg2png(
        url=svg_path,
        output_width=size,
        output_height=size
    )
    with open(output_path, 'wb') as f:
        f.write(png_data)

def generate_ico(svg_path, output_path):
    """Generate a Windows ICO file with multiple sizes"""
    # ICO files typically contain 16, 32, 48, and 256 pixel versions
    ico_sizes = [16, 32, 48, 256]
    images = []
    
    for size in ico_sizes:
        png_data = cairosvg.svg2png(
            url=svg_path,
            output_width=size,
            output_height=size
        )
        img = Image.open(io.BytesIO(png_data))
        images.append(img)
    
    # Save as ICO with all sizes
    images[0].save(
        output_path,
        format='ICO',
        sizes=[(img.width, img.height) for img in images],
        append_images=images[1:]
    )

def main():
    print("🎨 Generating CineVault icons from SVG...\n")
    
    # Check if SVG exists
    if not os.path.exists(SVG_PATH):
        print(f"❌ SVG file not found: {SVG_PATH}")
        return
    
    # Generate PNG files
    for size, filename in SIZES:
        output_path = os.path.join(ICONS_DIR, filename)
        generate_png(SVG_PATH, output_path, size)
        print(f"✓ Generated {filename} ({size}x{size})")
    
    # Generate ICO file
    print("\n📦 Generating Windows ICO file...")
    ico_path = os.path.join(ICONS_DIR, "icon.ico")
    generate_ico(SVG_PATH, ico_path)
    print("✓ Generated icon.ico (multi-size)")
    
    # ICNS note
    print("\n🍎 macOS ICNS file:")
    print("  Use online converter or png2icons tool")
    print("  See ICON_GENERATION.md for instructions")
    
    print("\n✨ Icon generation complete!")
    print(f"\nGenerated files in {ICONS_DIR}:")
    for _, filename in SIZES:
        print(f"  - {filename}")
    print("  - icon.ico")

if __name__ == "__main__":
    main()

File: tmp_rovodev_generate_icons.js
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';

const svgPath = 'src-tauri/icons/icon.svg';
const iconsDir = 'src-tauri/icons';

// Icon sizes we need to generate
const sizes = [
  { size: 32, name: '32x32.png' },
  { size: 128, name: '128x128.png' },
  { size: 256, name: '128x128@2x.png' },
  { size: 512, name: 'icon.png' },
  { size: 1024, name: 'icon@2x.png' }
];

// Linux icon sizes (for .desktop files and various icon themes)
const linuxSizes = [
  { size: 16, name: '16x16.png' },
  { size: 24, name: '24x24.png' },
  { size: 48, name: '48x48.png' },
  { size: 64, name: '64x64.png' },
  { size: 96, name: '96x96.png' },
  { size: 128, name: '128x128.png' },
  { size: 256, name: '256x256.png' },
  { size: 512, name: '512x512.png' }
];

async function generateIcons() {
  console.log('🎨 Generating CineVault icons from SVG...\n');

  try {
    // Read the SVG file
    const svgBuffer = fs.readFileSync(svgPath);

    // Generate PNG files for each size
    for (const { size, name } of sizes) {
      const outputPath = path.join(iconsDir, name);
      
      await sharp(svgBuffer)
        .resize(size, size)
        .png()
        .toFile(outputPath);
      
      console.log(`✓ Generated ${name} (${size}x${size})`);
    }

    // Generate ICO file (Windows icon with multiple sizes)
    console.log('\n📦 Generating Windows ICO file...');
    
    // For ICO, we need multiple sizes embedded (16, 32, 48, 256)
    const icoSizes = [16, 32, 48, 256];
    const icoBuffers = [];
    
    for (const size of icoSizes) {
      const buffer = await sharp(svgBuffer)
        .resize(size, size)
        .png()
        .toBuffer();
      icoBuffers.push(buffer);
    }
    
    // For simplicity, we'll use the 256px as the main ICO
    // A proper ICO would need a specialized library like png-to-ico
    await sharp(svgBuffer)
      .resize(256, 256)
      .png()
      .toFile(path.join(iconsDir, 'icon.ico'));
    
    console.log('✓ Generated icon.ico (256x256)');
    console.log('  Note: For a proper multi-size ICO, use a specialized tool');

    // Generate Linux icons
    console.log('\n🐧 Generating Linux icon sizes...');
    for (const { size, name } of linuxSizes) {
      const outputPath = path.join(iconsDir, name);
      
      // Skip if already generated (e.g., 128x128.png)
      if (fs.existsSync(outputPath)) {
        console.log(`↪ Skipped ${name} (already exists)`);
        continue;
      }
      
      await sharp(svgBuffer)
        .resize(size, size)
        .png()
        .toFile(outputPath);
      
      console.log(`✓ Generated ${name} (${size}x${size})`);
    }

    // Generate ICNS (macOS icon) - Sharp doesn't support this
    // We'll note this for the user
    console.log('\n🍎 macOS ICNS file:');
    console.log('  ⚠ ICNS generation requires specialized tools');
    console.log('  You can use: png2icons or iconutil (macOS only)');
    console.log('  Or use: https://cloudconvert.com/png-to-icns');

    console.log('\n✨ Icon generation complete!');
    console.log('\n📦 Windows icons:');
    sizes.forEach(({ name }) => console.log(`  - src-tauri/icons/${name}`));
    console.log(`  - src-tauri/icons/icon.ico`);
    console.log('\n🐧 Linux icons:');
    linuxSizes.forEach(({ name }) => console.log(`  - src-tauri/icons/${name}`));

  } catch (error) {
    console.error('❌ Error generating icons:', error);
    process.exit(1);
  }
}

generateIcons();
